srcpkgs/ngspice/template

```diff
@@ -1,28 +1,62 @@
 # Template file for 'ngspice'
 pkgname=ngspice
-version=33
+version=35
 revision=1
 build_style=gnu-configure
-configure_args="--with-readline=yes --with-ngshared --enable-xspice --enable-cider"
+configure_args="--with-readline=yes --enable-xspice --enable-cider"
+make_build_args="-C .build_lib"
 hostmakedepends="bison byacc"
 makedepends="readline-devel libX11-devel libXaw-devel"
+checkdepends="perl"
 short_desc="Mixed Mode Mixed Level Circuit Simulator based on Spice3F5"
 maintainer="Toyam Cox <Vaelatern@voidlinux.org>"
 license="BSD-3-Clause"
 homepage="http://ngspice.sourceforge.net"
 distfiles="${SOURCEFORGE_SITE}/ngspice/ng-spice-rework/${version}/${pkgname}-${version}.tar.gz"
-checksum=b99db66cc1c57c44e9af1ef6ccb1dcbc8ae1df3e35acf570af578f606f8541f1
+checksum=c1b7f5c276db579acb3f0a7afb64afdeb4362289a6cab502d4ca302d6e5279ec
+
+# Configure can handle building ngspice as a library or as a binary, but not both at once.
+# --with-ngshared builds the library, but not binary, and readline should not be configured for the library,
+# see: https://github.com/imr/ngspice/commit/b86c85f85bbba6e45dc030af3e853edf8b9cfa3d
+pre_configure() {
+	install -d .build_lib
+	cd .build_lib
+	../configure --with-ngshared ${configure_args/--with-readline=yes/}
+}
+
+# library built during first do_build, now build the binary
+post_build() {
+	make_build_args= do_build
+}
+
+# Starting with v34, some tests fail with "No compatability mode selected!"
+# Solution is to add this line to a .spiceinit file in the home directory,
+# see: https://sourceforge.net/p/ngspice/bugs/554/#b513
+pre_check() {
+	export HOME="${PWD}/.home_dir"
+	install -d "$HOME"
+	printf 'set ngbehavior=mc\n' > "${HOME}/.spiceinit"
+}
 
 post_install() {
 	vlicense COPYING
+	cd .build_lib
+	do_install
 }
```

Comment:
And:
```suggestion
do_install() {
	make -C build-lib install
	make -C build install
	vlicense COPYING
}
```

Reply:
It does seem clearer this way.  I kind of liked not touching do_install, etc., to keep defaults and things, but maybe that's just obscuring things.`